homeassistant/components/lyric/config_flow.py

```diff
@@ -0,0 +1,152 @@
+"""Config flow for Honeywell Lyric."""
+import asyncio
+import logging
+
+import async_timeout
+from lyric import Lyric
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.config_entries import ConfigFlow
+from .const import (AUTH_CALLBACK_NAME, AUTH_CALLBACK_PATH, DOMAIN,
+                    CONF_CLIENT_ID, CONF_CLIENT_SECRET, CONF_LYRIC_CONFIG_FILE)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class LyricFlowHandler(ConfigFlow):
+    """Handle a Lyric config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize Lyric flow."""
+        self.client_id = None
+        self.client_secret = None
+        self.code = None
+        pass
+
+    async def _show_setup_form(self, errors=None):
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_CLIENT_ID): str,
```

Comment:
Client ID and secret are not allowed to be asked via a config flow and need to be set via `configuration.yaml`. 

Example in Toon:
 - [config schema](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/toon/__init__.py#L43-L56)